util/openssl-update-copyright

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+#
+# Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+
+myname="$(basename $0)"
+
+this_year="$(date '+%Y')"
+some_year="[12][0-9][0-9][0-9]"
+
+copyright_owner="The OpenSSL Project"
+
+search="Copyright \(([cC]) \)\?\(${some_year}\)\(-${some_year}\)\? ${copyright_owner}"
+replace="Copyright \1\2-${this_year} ${copyright_owner}"
+
+
+function usage() {
+	cat >&2 <<EOF
+usage: $myname [-h|--help] [file|directory] ...
+
+Updates the year ranges of all OpenSSL copyright statements in the given
+files or directories. (Directories are traversed recursively.)
+EOF
+}
+
+if [ $# -eq 0 ]; then
+	usage
+	exit 0
+fi
+
+
+for arg in "$@"; do
+	case $arg in
+		-h|--help)
+			usage
+			exit 0
+			;;
+		-*)
+			echo -e "illegal option: $arg\n" >& 2
+			usage
+			exit 1
+			;;
+		*)
+			if [ -f "$arg" ]; then
+				sed -i "s/${search}/${replace}/g" "$arg"
```

Comment:
This form of ```sed -i``` expression with no extension for a backup file is something of a GNU-ism, and is not necessarily portable to BSD sed.
This is probably not a fatal flaw, since this is expected to be a maintainer-only tool and we can deal with it; I just wanted to note the nonportability.

Reply:
As far as I googled it, the only way to satisfy GNU and BSD sed is to add the backup extension without a space, like `-i.bak`, right?

It's a pity that your comments came too late for this PR. If the two portability issues (this one and the shebang line) disturb you then feel free to change them, I won't mind. Since I don't have a BSD system lying around, I can't test it myself.

Reply:
I think that's right about ```-i.bak```, yes.

But I do not plan to submit a follow-up; so far as I know all the people who might need to run this script have access to a system where it should work without modification.